### PR TITLE
dell-kestrel: Do not show 'Device has been removed' as the device error

### PR DESCRIPTION
**Pull request by hughsie**
Add ONLY_WAIT_FOR_REPLUG to all final devices or subclasses.

Fixes https://github.com/fwupd/fwupd/issues/8183

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — DISMISSED** [review]

**Review by CragW — DISMISSED** [review]

**Comment by CragW** [comment]
Do we need this flag on `mst` and `usb4` devices?

**Comment by superm1** [comment]
Usb4 shouldn't need it, it's got it's own end point that really disappears doesn't it?

**Comment by CragW** [comment]
> Usb4 shouldn't need it, it's got it's own end point that really disappears doesn't it?

I can see the `usb4` device showing the same error state occasionally.

**Comment by superm1** [comment]
> > Usb4 shouldn't need it, it's got it's own end point that really disappears doesn't it?
> 
> I can see the `usb4` device showing the same error state occasionally. 

Oh I see. @hughsie sounds sensible to me, agree?

**Comment by CragW** [comment]
> > Usb4 shouldn't need it, it's got it's own end point that really disappears doesn't it?
> 
> I can see the `usb4` device showing the same error state occasionally.

vmm9 added the flag for mst, but I think we can do a safe in this plugin.

**Review by CragW — APPROVED** [review]

File: plugins/dell-kestrel/fu-dell-kestrel-plugin.c [hunk 1]
```diff
@@ -256,6 +256,7 @@ fu_dell_kestrel_plugin_config_mst_dev(FuPlugin *plugin)
 	/* flags */
 	fu_device_add_private_flag(device_mst, FU_DEVICE_PRIVATE_FLAG_EXPLICIT_ORDER);
 	fu_device_add_private_flag(device_mst, FU_DEVICE_PRIVATE_FLAG_SKIPS_RESTART);
+	fu_device_add_private_flag(device_mst, FU_DEVICE_PRIVATE_FLAG_ONLY_WAIT_FOR_REPLUG);
```
Comment (hughsie) [inline plugins/dell-kestrel/fu-dell-kestrel-plugin.c:259]: this is already set in the fu_intel_usb4_device_init -- no?

File: plugins/dell-kestrel/fu-dell-kestrel-plugin.c [hunk 2]
```diff
@@ -296,6 +297,7 @@ fu_dell_kestrel_plugin_device_registered(FuPlugin *plugin, FuDevice *device)
 			return;
 		}
 		fu_device_add_private_flag(device, FU_DEVICE_PRIVATE_FLAG_EXPLICIT_ORDER);
+		fu_device_add_private_flag(device, FU_DEVICE_PRIVATE_FLAG_ONLY_WAIT_FOR_REPLUG);
```
Comment (hughsie) [inline plugins/dell-kestrel/fu-dell-kestrel-plugin.c:300]: this is already set in the fu_intel_usb4_device_init -- no?

**Comment by superm1** [comment]
> I'm actually wondering if FU_DEVICE_PRIVATE_FLAG_ONLY_WAIT_FOR_REPLUG is a footgun we want to remove in 2.0.x -- i.e. flip it around so the weirdo plugins have to set the opposite action -- or just remove it completely as IIRC it was only for plugins that forgot to set FWUPD_DEVICE_FLAG_WAIT_FOR_REPLUG, or where the device restart happened when we couldn't predict.

Sounds reasonable. It's a private flag, can easily flip or remove.

File: plugins/dell-kestrel/fu-dell-kestrel-dpmux.c [hunk 3]
```diff
@@ -87,6 +87,7 @@ fu_dell_kestrel_dpmux_init(FuDellKestrelDpmux *self)
 	fu_device_set_version_format(FU_DEVICE(self), FWUPD_VERSION_FORMAT_QUAD);
 	fu_device_add_private_flag(FU_DEVICE(self), FU_DEVICE_PRIVATE_FLAG_EXPLICIT_ORDER);
 	fu_device_add_private_flag(FU_DEVICE(self), FU_DEVICE_PRIVATE_FLAG_USE_PROXY_FOR_OPEN);
+	fu_device_add_private_flag(FU_DEVICE(self), FU_DEVICE_PRIVATE_FLAG_ONLY_WAIT_FOR_REPLUG);
```
Comment (CragW) [inline plugins/dell-kestrel/fu-dell-kestrel-dpmux.c:90]: I've retested couple times, summarized the impacted devices as following (which I think should add this flag):
* ec
* rts usbs
* rmm

The following devices have added this flag in their own plugin
* mst
* usb4

The following devices never show this error, as they are probed by ec dynamically:
* dpmux
* ilan
* pkg
* pd
* wtpd

Comment (hughsie) [reply plugins/dell-kestrel/fu-dell-kestrel-dpmux.c:90]: That looks pretty good; can you edit my original patch to only set the flag for ec, rts and rmm please.

File: plugins/dell-kestrel/fu-dell-kestrel-plugin.c (hunk 1 above)
Comment (hughsie) [reply plugins/dell-kestrel/fu-dell-kestrel-plugin.c]: Nah, I want to get rid of FU_DEVICE_PRIVATE_FLAG_ONLY_WAIT_FOR_REPLUG in the next couple of releases anyway.

**Pull request merged**